Dockerfile

```diff
@@ -0,0 +1,32 @@
+# syntax=docker/dockerfile:experimental
+
+FROM opensuse/tumbleweed:latest
+RUN zypper ref && zypper up -y
+#RUN zypper in -y azure-cli
+
+# way suggested on https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-linux?pivots=zypper
+RUN zypper install -y tar gzip unzip curl python310-pip openssh
+RUN rpm --import https://packages.microsoft.com/keys/microsoft.asc
+RUN zypper addrepo --name 'Azure CLI' --check https://packages.microsoft.com/yumrepos/azure-cli azure-cli
+RUN zypper install --from azure-cli -y azure-cli
+RUN zypper clean --all
+
+
+WORKDIR /root
+RUN curl https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-cli-387.0.0-linux-x86_64.tar.gz -o google-cloud-cli.tar.gz
```

Comment:
Same here.  This `RUN will create a layer with the size of the tarball.  You can run those command concatenated and then remove the tarball and you would get a smaller size.